pkg/kubectl/resource_printer.go

```diff
@@ -95,27 +96,55 @@ func (fn ResourcePrinterFunc) PrintObj(obj runtime.Object, w io.Writer) error {
 	return fn(obj, w)
 }
 
-// VersionedPrinter takes runtime objects and ensures they are converted to a given API version
-// prior to being passed to a nested printer.
-type VersionedPrinter struct {
-	printer   ResourcePrinter
+type ObjectPreprocessor interface {
+	Process(runtime.Object) (runtime.Object, error)
+}
+
+type chainObjectPreprocessor struct {
```

Comment:
> I'd rather have a slice of ObjectPreprocessors

To clarify
```
type ObjectPreprocessors []ObjectPreprocessor

func (p ObjectPreprocessors) Process(obj runtime.Object) (runtime.Object, error) {
    currObj := obj
    for _, curr := range p{
        currObj, err = curr.Process(currObj)
        if err != nil{
            return err
        }
    }

    return currObj, nil
}
```

slice instead of linkedlist.